test/test_linalg.py

```diff
@@ -963,6 +963,147 @@ def get_value_type(dtype):
                         with self.assertRaisesRegex(RuntimeError, r'provided dtype must match dtype of result'):
                             torch.linalg.norm(input, ord=ord, keepdim=keepdim, dtype=dtype, out=result)
 
+    # This test confirms that torch.linalg.vector_norm's dtype argument works
+    # as expected, according to the function's documentation
+    @skipCUDAIfNoMagma
```

Comment:
This is a very thorough test. If a reference is created (as suggested below) then could that later test be expanded by also enumerating over dtype and just validating that the behavior is the same as linalg.norm's?

Reply:
Ah yeah, that's a really good simplification